recipes/lmdb/all/conanfile.py

```diff
@@ -12,8 +12,16 @@ class lmdbConan(ConanFile):
     description = "Fast and compat memory-mapped key-value database"
     topics = ("LMDB", "database", "key-value", "memory-mapped")
     settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
+    options = {
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "disable_robust_mutex": [True, False], # switches MDB_USE_ROBUST
+    }
+    default_options = {
+            "shared": False,
+            "fPIC": True,
+            "disable_robust_mutex": False,
```

Comment:
Wouldn't it be nicer to name this option `enable_robust_mutex` (and default to True)?
It's easier to reason about enabling an options instead of double negating an option to enable someting.

`enable_robust_mutex=True` vs `disable_robust_mutex=False`
It's perhaps a personal taste, but of the two choices, I prefer the first.

Reply:
or just `robust_mutex`

Reply:
It could be nicer if we would use it every time...
Normally LMDB as part of it's implementation detects (glibc version/OS version/...) if it would use robust mutex. Most of the times it detects it correct and we don't wanna need to do anything.
This option is kind-a "force disable robust mutex" independently from what lmdb wants to use, so I'd keep it as it is if possible

Reply:
I strongly prefer "positive" names too, so I'd rather have an option named `enable_robust_mutex` which is `True` by default

Reply:
We need to keep automatic packet setting in place, so how we make it then?
* positive case -> MDB_USE_ROBUST=1
* negative case -> MDB_USE_ROBUST=0
* normal/auto case -> no MDB_USE_ROBUST define

My proposal would be then:
"enable_robust_mutex": [Auto, True, False]
and default_options = {
...
"enable_robust_mutex": Auto
}

@ericLemanissier @madebr  @SpaceIm  what do you think?

Reply:
We don't necessarily need automatic detection in place, as long as we provide a sufficiently similar experience.
I looked through [`mdb.c`](https://github.com/LMDB/lmdb/blob/mdb.master/libraries/liblmdb/mdb.c) and found out robust mutexes are only enabled by default on FreeBSD and newer glibc (not Android).

So doing the following would work too:
```python
options = {
    "enable_robust_mutex": [True, False],
}
default_options = {
    "enable_robust_mutex": True,  # default modified in config_options
}
def config_options(self):
    if self.settings.os in ("Linux", "FreeBSD"):
        self.options.enable_robust_mutex = True
    else:
        self.options.enable_robust_mutex = False
```

Reply:
@madebr  @ericLemanissier should be fixed now :-) Thanks for the tip

Reply:
oh... I confused smth with push